ui/src/app/store/manage-collectiom/reducers.ts

```diff
@@ -160,10 +160,19 @@ const reducer: ActionReducer<CollectionEntityState> = createReducer(
   }),
 
   on(getSubjectExamplesFail, state => ({ ...state, isCollectionPending: false })),
-  on(addFileToCollection, (state, { url, file, subject }) => ({
-    ...state,
-    collection: [...state.collection, { url, file, subject, status: CircleLoadingProgressEnum.OnHold }],
-  })),
+
+  on(addFileToCollection, (state, { url, file, subject }) => {
+    const collectionCopy = [
+      ...state.collection.filter(item => item.status !== CircleLoadingProgressEnum.Failed && item.subject === state.subject),
+    ];
+
+    const failedCollection = [...state.collection.filter(item => item.status === CircleLoadingProgressEnum.Failed && item.url !== url)];
+
+    return {
+      ...state,
+      collection: [...collectionCopy, ...failedCollection, { url, file, subject, status: CircleLoadingProgressEnum.OnHold }],
```

Comment:
we can also avoid assigning memory to arrays which we get rid of right away. how about something like this?
```
return {
    ...state,
    collection: [
        ...state.collection.filter(item => 
            (item.status !== CircleLoadingProgressEnum.Failed && item.subject === state.subject) // not failed
            || (item => item.status === CircleLoadingProgressEnum.Failed && item.url !== url) // failed
        ), { url, file, subject, status: CircleLoadingProgressEnum.OnHold }
    ],
}
```

Reply:
fixed

Reply:
fixed